### ISSUE TITLE
working with unitized numpy arrays in plasma parameters functions

### ISSUE THREAD

**Issue by samurai688** [PlasmaPy/PlasmaPy#469]
Life seems pretty good when we can work with unitized numpy arrays in plasma parameter functions. I guess these are technically Quantity objects, i.e. as created by `a = np.array([1,2,3,4]) * u.m` A good current example is the Debye length function.  You can pass it big unitized numpy arrays of densities and temperatures, and it will give you the elementwise result of debye length.  It seems fast for big arrays just like you'd hope from numpy. I haven't tested it too rigorously but it seems to work really well out of the box.

Some of our current functions will fail when given numpy arrays. I'm looking at you, gyroradius, collision_frequency, impact_parameter, and some others.  It seems like most of the problems show up when we have an if statement like 

```
if bPerp > lambdaBroglie:
    bmin = bPerp
else:
    bmin = lambdaBroglie
```

If we try to give it inputs such that `bPerp` arrives as a unitized numpy array, it will give an exception "the truth value of an array is ambiguous, use .any() or .all()" or some such.  A wrinkle of this one in particular, we want to get the maxes into our `bmin` result, which would then be a numpy array. I would typically do `np.max((bPerp, lambdaBroglie), axis=0)` but that seems to ignore the units, and drop them, for good measure.  The comparison `bPerp > lambdaBroglie` seems to work in a nice unit-aware way, so I guess we can work with the boolean array we get from that...

So anyway,
1) do you think we should support these objects throughout our functions?
2) any clever ideas to handle the issues of the type above?
3) general thoughts?

**Comment by namurphy** [comment]
Yes, this is an important issue and we should allow all of our plasma parameters functions to accept and return `astropy.units.Quantity` objects.  We should also make sure that there are tests for each of the plasma parameters functions that take in `Quantity` objects.

Cross-reference for another issue about `Quantity` objects that we should keep in mind: #59

**Comment by lemmatum** [comment]
I'll add that sometimes arrays with units are clunky for things like list comprehensions. Perhaps we should add some convenience functions for handling these cases?

**Comment by StanczakDominik** [comment]
Closed by #477, #480, #485 and #492 and thanks @samurai688 once again! :D
